Documentation for new setStatusMessage action and presenceStatusMessage type (#19239)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/presence-setstatusmessage.md

---
title: "presence: setStatusMessage"

> description: "Set the Teams Presence Status Message for a user"

Reviewer: ```suggestion
description: "Set presence status message for a user.".
```

author: "afedorov"
ms.localizationpriority: medium
doc_type: apiPageType
ms.prod: "cloud-communications"
---

# presence: setStatusMessage

Namespace: microsoft.graph

[!INCLUDE [beta-disclaimer](../../includes/beta-disclaimer.md)]

> "Set presence status message for a user. An optional expiration date and time can be supplied.

Reviewer: ```suggestion
Sets presence status message for a user. An optional expiration date and time can be supplied.
```

> Set the Teams Presence Status Message for a user. Optional expiration date and time can be supplied as well.

Reviewer: ```suggestion
"Set presence status message for a user. An optional expiration date and time can be supplied.
```

## Permissions
The following permission is required to call the API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

| Permission type                        | Permissions (from least to most privileged) |
| :------------------------------------- | :------------------------------------------ |
| Delegated (work or school account)     | Presence.ReadWrite                          |
| Delegated (personal Microsoft account) | Not Supported.                              |
| Application                            | Not Supported.                      |

## HTTP Request
<!-- { "blockType": "ignored" } -->
```http
POST /users/{userId}/presence/setStatusMessage
```

## Request headers
| Name          | Description                 |
| :------------ | :-------------------------- |
| Authorization | Bearer \{token\}. Required.   |
| Content-Type  | application/json. Required. |

## Request body

In the request body, provide a JSON object with the following parameters.

| Parameter       | Type            |  Description                                 |
| --------------- | --------------- |------------------------------------------- |

> | `statusMessage` | [microsoft.graph.presenceStatusMessage](../resources/presenceStatusMessage.md) | Status message that can be set to display a Teams Presence Status Message of a user. |

Reviewer: ```suggestion
| `statusMessage` | [microsoft.graph.presenceStatusMessage](../resources/presenceStatusMessage.md) |It can be set to display the Teams presence status message of a user. |
```

The following permission is required to call the API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

| Permission type                        | Permissions (from least to most privileged) |
| :------------------------------------- | :------------------------------------------ |
| Delegated (work or school account)     | Presence.ReadWrite                          |
| Delegated (personal Microsoft account) | Not Supported.                              |
| Application                            | Presence.ReadWrite.All                      |

## HTTP Request
<!-- { "blockType": "ignored" } -->
```http
POST /users/{userId}/presence/setStatusMessage
```

## Request headers
| Name          | Description                 |
| :------------ | :-------------------------- |
| Authorization | Bearer \{token\}. Required.   |
| Content-Type  | application/json. Required. |

## Request body

In the request body, provide a JSON object with the following parameters.

| Parameter       | Type            |  Description                                 |
| --------------- | --------------- |------------------------------------------- |

> | `statusMessage` | [microsoft.graph.presenceStatusMessage](../resources/presenceStatusMessage.md) | Status message that can be set to display a text message in presence. |

Reviewer 2: wondering if the term "text message" might confusing. Can we just call it the Teams' Presence Status Message

Reviewer 2: and keep the naming consistent in all the PR docs

## Response
If successful, this method returns a `200 OK` response code.

## Examples

### Example 1: Set status message with expiry date

> The following request sets the user status message as "Hey I'm currently in a meeting." for user `fa8bf3dc-eca7-46b7-bad1-db199b62afc3`, with the expiration on 2022-10-18 at 17:05:33.2079781 Pacific Standard Time.

Reviewer 2: pl double check that the id is not a real user and a random guid

Reviewer 2: in other places in this PR too

Author: This id is already used in setPreferredPresence, I took it from there.
How could I check that id is real user id or not?

Reviewer 2: if it is the same id, then it should have already been checked. There is no way for us to check it now

###### [PowerShell](#tab/powershell)
[!INCLUDE [sample-code](../includes/snippets/powershell/setstatusmessage-powershell-snippets.md)]
[!INCLUDE [sdk-documentation](../includes/snippets/snippets-sdk-documentation-link.md)]

###### [PHP](#tab/php)
[!INCLUDE [sample-code](../includes/snippets/php/setstatusmessage-php-snippets.md)]
[!INCLUDE [sdk-documentation](../includes/snippets/snippets-sdk-documentation-link.md)]

---

#### Response

<!-- {
  "blockType": "response",
  "truncated": true
} -->
```http
HTTP/1.1 200 OK
```

### Example 2: Set status message: xml sanitization

The first request sets the user status message as "Hey I'm currently in a meeting." for user `fa8bf3dc-eca7-46b7-bad1-db199b62afc3`. Then user's self presence with status message is obtained using a [getPresence](presence-get.md) request.

#### Set status message: request

##### [HTTP](#tab/http)
<!-- {
  "blockType": "request",
  "name": "setstatusmessage-xml-sanitization"
}-->
```msgraph-interactive
POST https://graph.microsoft.com/beta/users/fa8bf3dc-eca7-46b7-bad1-db199b62afc3/presence/setStatusMessage
Content-Type: application/json

{
  "statusMessage": {
    "message": {

>       "content": "<html>Hello world a<b</html>",

Reviewer 2: IMO we should give high level examples and do not need to give corner cases as examples.

Can we check with the UPS team if we they even want us to document the sanitization aspects. I am wondering if we are going to encourage bad actors if we detail too much on our sanitization strategy

Author: The motivation for this example was to visualize what xml sanitization could be to bring more clarity to API behavior.
That could be useful for those customers who want to encode some data in the presence note.

I posted a question to UPS team as well.

Could you please clarify "bad actors"?
